tests/postgres_tests/test_operations.py

```diff
@@ -459,6 +459,24 @@ def test_writer(self):
             "),",
         )
 
+    def test_reduce_create_remove(self):
+        self.assertOptimizesTo(
+            [
+                CreateCollation(
+                    "sample_collation",
+                    "und-u-ks-level2",
+                    provider="icu",
+                    deterministic=False,
+                ),
+                RemoveCollation(
+                    "sample_collation",
+                    # Different locale
+                    "de-u-ks-level1",
+                ),
+            ],
+            [],
+        )
+
```

Comment:
Should the reverse (they remove and add it back in) also reduce to nothing?
This fails currently
```suggestion
        self.assertOptimizesTo(
            [
                RemoveCollation("sample_collation", locale="C"),
                CreateCollation("sample_collation", locale="C"),
            ],
            [],
        )
```

Reply:
As far as I'm aware, this cannot happen in a valid operations list. You cannot remove something that was not created in advance.

Reply:
Confirmed this isn't something we support with other things like `RemoveField` then `AddField`

The scenario I was thinking was squashing some migrations in an app so I made a model with a field, makemigrations (`0001` creates the model with a field), remove field makemigrations (`0002` removes a field), add it back makemigrations (`0002` adds the field back) 
run `python manage.py squashmigrations myapp 0002 0003` and it says it can't optimize (I expect it can reduce to nothing). Optimizes as expected when doing `python manage.py squashmigrations myapp 0003`

So, long story short, this is separate to this PR and there might be reasons why we've decided not to support this

Reply:
IMO, we cannot optimize such scenario because there is no guarantee that the field created in `0003` is the same as the field created in `0001`.

Reply:
For example:
- `0001` - `CreateCollation("sample_collation", locale="C"),`
- `0002` - `RemoveCollation("sample_collation", locale="C")`
- `0003` - `CreateCollation("sample_collation", locale="und-u-ks-level2", provider="icu", deterministic=False)`

We can optimize `0001`-`0002` to no-op, but we cannot optimize `0002`-`0003` to no-op as we would end with `"sample_collation", locale="C"`.

Reply:
Exactly. And even if they are the same, migrations 0001-0003 can be squashed together, at which point the optimizer would remove the create/remove pair.